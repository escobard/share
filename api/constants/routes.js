const routes = {
    port: process.env.PORT || 4000,
    health: '/health',
    greetings: '/greetings',
    makeDonation: '/makeDonation',
    makeDonationStatus: '/makeDonationStatus',
    fetchDonation: '/fetchDonation'
};

module.exports = routes;
