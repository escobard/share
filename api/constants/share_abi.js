const abi = [
  {
    "constant": true,
    "inputs": [
      {
        "name": "",
        "type": "address"
      }
    ],
    "name": "lotteryEntrees",
    "outputs": [
      {
        "name": "",
        "type": "address"
      }
    ],
    "payable": false,
    "stateMutability": "view",
    "type": "function"
  },
  {
    "inputs": [
      {
        "name": "_ownable",
        "type": "address"
      },
      {
        "name": "_charityRole",
        "type": "address"
      },
      {
        "name": "_lotteryRole",
        "type": "address"
      }
    ],
    "payable": false,
    "stateMutability": "nonpayable",
    "type": "constructor"
  },
  {
    "constant": false,
    "inputs": [
      {
        "name": "_lottery",
        "type": "address"
      },
      {
        "name": "_charity",
        "type": "address"
      }
    ],
    "name": "initiateContract",
    "outputs": [],
    "payable": true,
    "stateMutability": "payable",
    "type": "function"
  },
  {
    "constant": false,
    "inputs": [],
    "name": "makeDonation",
    "outputs": [],
    "payable": true,
    "stateMutability": "payable",
    "type": "function"
  },
  {
    "constant": true,
    "inputs": [],
    "name": "fetchDonationID",
    "outputs": [
      {
        "name": "",
        "type": "uint256"
      }
    ],
    "payable": false,
    "stateMutability": "view",
    "type": "function"
  },
  {
    "constant": true,
    "inputs": [
      {
        "name": "_id",
        "type": "uint256"
      }
    ],
    "name": "fetchDonation",
    "outputs": [
      {
        "name": "owner",
        "type": "address"
      },
      {
        "name": "lottery",
        "type": "address"
      },
      {
        "name": "charity",
        "type": "address"
      },
      {
        "name": "donor",
        "type": "address"
      },
      {
        "name": "amount",
        "type": "uint256"
      },
      {
        "name": "charityAmount",
        "type": "uint256"
      },
      {
        "name": "lotteryAmount",
        "type": "uint256"
      },
      {
        "name": "ownerAmount",
        "type": "uint256"
      },
      {
        "name": "id",
        "type": "uint256"
      }
    ],
    "payable": false,
    "stateMutability": "view",
    "type": "function"
  },
  {
    "constant": true,
    "inputs": [],
    "name": "isInitialized",
    "outputs": [
      {
        "name": "",
        "type": "bool"
      }
    ],
    "payable": false,
    "stateMutability": "view",
    "type": "function"
  }
];

module.exports = abi;
