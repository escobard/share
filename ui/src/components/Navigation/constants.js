
export const items = [{ key: "home", active: true, name: "share v1" }];

