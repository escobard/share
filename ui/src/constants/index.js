const makeDonationFields = [
  {
    name: "addressPub",
    label: "Address Public",
    placeholder: "Donor's public address",
    value: "",
    error: false
  },
  {
    name: "addressPriv",
    label: "Address Private",
    placeholder:
      "Donor's private address - only used to SIGN raw transactions within API, not stored or cached",
    value: "",
    error: false
  },
  {
    name: "amount",
    label: "Amount",
    placeholder: "Donation amount in ether",
    value: "",
    error: false
  }
];

const fetchDonationFields = [
  {
    name: "address",
    label: "Address Public",
    placeholder: "Enter user's public address",
    value: "",
    error: false
  },
  {
    name: "id",
    label: "Donation ID",
    placeholder: "Enter the Donation ID",
    value: "",
    error: false
  }
];

const environment =
  process.env.NODE_ENV === "production" ? "heroku" : "development";

console.log('environment', environment);

const apiRoot = environment === "heroku" ? "https://share-controller.herokuapp.com" : "http://localhost:4000";

console.log('root', apiRoot)

const apiRoutes = {
  makeDonation: `${apiRoot + "/makeDonation"}`,
  fetchDonation: `${apiRoot + "/fetchDonation"}`
};

export { makeDonationFields, fetchDonationFields, apiRoutes };
