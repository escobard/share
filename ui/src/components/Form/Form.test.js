import React from "react";
import { shallow } from "enzyme";
import DynamicForm from "./index";
import { fetchDonationFields, makeDonationFields } from "../../constants";

describe("Form snapshot renders", () => {
  it("should render makeDonation form correctly", () => {
    const component = shallow(<DynamicForm fields={makeDonationFields} />);

    expect(component).toMatchSnapshot();
  });
  it("should render fetchDonation form correctly", () => {
    const component = shallow(<DynamicForm fields={fetchDonationFields} />);

    expect(component).toMatchSnapshot();
  });
});
