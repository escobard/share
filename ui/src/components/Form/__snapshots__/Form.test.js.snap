// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form snapshot renders should render fetchDonation form correctly 1`] = `
<Fragment>
  <Form
    as="form"
  >
    <Message />
    <FormGroup
      widths="equal"
    >
      <FormInput
        as={[Function]}
        control={[Function]}
        key="0"
        label="Address Public"
        name="address"
        onChange={[Function]}
        placeholder="Enter user's public address"
      />
      <FormInput
        as={[Function]}
        control={[Function]}
        key="1"
        label="Donation ID"
        name="id"
        onChange={[Function]}
        placeholder="Enter the Donation ID"
      />
    </FormGroup>
    <FormField
      control={[Function]}
      onClick={[Function]}
    >
      Submit
    </FormField>
  </Form>
</Fragment>
`;

exports[`Form snapshot renders should render makeDonation form correctly 1`] = `
<Fragment>
  <Form
    as="form"
  >
    <Message />
    <FormGroup
      widths="equal"
    >
      <FormInput
        as={[Function]}
        control={[Function]}
        key="0"
        label="Address Public"
        name="addressPub"
        onChange={[Function]}
        placeholder="Donor's public address"
      />
      <FormInput
        as={[Function]}
        control={[Function]}
        key="1"
        label="Private Key"
        name="privKey"
        onChange={[Function]}
        placeholder="Donor's private address - only used to SIGN raw transactions within API, not stored or cached"
      />
      <FormInput
        as={[Function]}
        control={[Function]}
        key="2"
        label="Amount"
        name="amount"
        onChange={[Function]}
        placeholder="Donation amount in ether"
      />
    </FormGroup>
    <FormField
      control={[Function]}
      onClick={[Function]}
    >
      Submit
    </FormField>
  </Form>
</Fragment>
`;
